Tests/BotTests/GitHub/GitHubEventsTests.swift

```diff
@@ -99,9 +141,14 @@ extension GitHubEventsTests {
             _ type: T.Type,
             using scheduler: QueueScheduler
         ) -> SignalProducer<T, AnyError> where T: Decodable {
-            switch body as? T {
-            case let .some(value):
-                return SignalProducer(value: value)
+            switch data {
+            case let .some(data):
+                let decoder = JSONDecoder()
+                do {
+                    return SignalProducer(value: try decoder.decode(T.self, from: data))
+                } catch {
+                    return SignalProducer(error: AnyError(DecodeError.invalid))
+                }
```

Comment:
[nit] just a style question, why the `switch` and not a `guard let data = data else { return SignalProducer(error: AnyError(DecodeError.invalid)) }` early return?

Reply:
No reason really, I've changed it Olivier thanks.